src/AWS.Logger.Core/Core/AWSLoggerCore.cs

```diff
@@ -229,15 +229,22 @@ public void Flush()
                 }
             }
         }
-
+        
```

Comment:
In the `LogEventTransmissionSetup` there needs to be a null check when checking the log group collection after calling `DescribeLogGroupsAsync`.

Also the the `PutLogEventsRequest _request` field in `AWSLoggerCore` is initialized the `LogEvents` collection is never initialized. I think you need to change the line 
```
public PutLogEventsRequest _request = new PutLogEventsRequest();
```
to
```
public PutLogEventsRequest _request = new PutLogEventsRequest {LogEvents = new List<InputLogEvent>() };
```

Not related to your change but I don't know why the `_request` field is public. Can you try and make that private and see if there are any issues. My quick scan shows it only used in `AWSLoggerCore`.

Reply:
I addressed your comments.
As for the public field, it is defined inside the `LogEventBatch` class and then accessed from `AWSLoggerCore`. So it needs to be public.